lib/new_relic/helper.rb

```diff
@@ -40,5 +44,41 @@ def instance_methods_include?(klass, method_name)
     def time_to_millis(time)
       (time.to_f * 1000).round
     end
+
+    def run_command(command)
+      executable = command.split(' ').first
+      unless executable_in_path?(executable)
+        raise NewRelic::CommandExecutableNotFoundError.new("Executable not found: '#{executable}'")
+      end
+
+      exception = nil
+      begin
+        output, status = Open3.capture2e(command)
+      rescue => exception
+      end
+
+      if exception || !status.success?
+        message = exception ? "#{exception.class} - #{exception.message}" : output
+        raise NewRelic::CommandRunFailedError.new("Failed to run command '#{command}': #{message}")
+      end
+
+      output.chomp
+    end
+
+    # TODO: Open3 defers the actual excecution of a binary to Process.spawn,
+    #       which will raise an Errno::ENOENT exception for a file that
+    #       cannot be found. We might want to take the time to evaluate
+    #       relying on that Process.spawn behavior instead of checking for
+    #       existence ourselves. We'd need to see what it does, how efficient
+    #       it is, if it differs in functionality between Ruby versions and
+    #       operating systems, etc.
```

Comment:
We talked a bit about todos recently, but not quite what to do with them! Should we make issues? Leave them for analysis at the beginning of a bugsmash?

Reply:
Issues would be good. The issue could reference the class and method where the TODO lives. It would be nice to have the TODO text reference the issue as well so that they're linked bidirectionally but that requires a PR. Maybe we can do some bulk issue creation and then produce a single PR with all of the updated TODO comments.

Reply:
I like it! I can make a ticket for bulk TODO ticket making 😉